### ISSUE TITLE
Where is the backup wallet stored? I can't find it...

### ISSUE THREAD

**Issue by megmaltese** [bitpay/copay#4457]
As per title... wanted to make a backup and copy it to MEGA but can't find it.

**Comment by gabrielbazan7** [comment]
Wallet preferences -> backup (12 words backup) or Wallet preferences -> preferences advanced -> export wallet (file/text backup)

**Comment by megmaltese** [comment]
Hi,

thank you for the answer and for the wallet too!

One stupid question: how can the wallet recover my keys from 12 words?
Those words are generated at the very beginning, so in no way they are
representing the actual status of my funds in the future.
So where are my funds kept track of if I lose my wallet and recover it with
the 12 words?
Or do the words change each time I use the wallet? I can't understand how
this HD thing work at all.

Marco E. G. Maltese

On 19 June 2016 at 20:05, Gabriel Edgardo Bazán <notifications@github.com>
wrote:

> Wallet preferences -> backup (12 words backup) or Wallet preferences ->
> preferences advanced -> export wallet (file/text backup)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/bitpay/copay/issues/4457#issuecomment-227011356>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/AHxHsTZXiaz67nQ1l6eBqTTG0OJRB94tks5qNYTPgaJpZM4I5Nky>
> .
>


<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DDB4FAA8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by dabura667** [comment]
Bitcoin private keys are large random numbers.

Those twelve words have more randomness than the number of grains of sand on earth.

There is a formula to convert any phrase into a large number. This is the first private key.

There is another formula that changes the previous private key into another, seemingly random private key. This generates the second, third, etc. private key.

If you input the same phrase, you get the exact same chain of keys in the exact same order.

The app just generates the addresses in order.

It's a lot more complicated, but that's the ELI5.

**Comment by matiu** [comment]
To add to @dabura667 excellent response ... once you have you master key, you can scan the blockchain and derive the rest of the "child" keys, so you you can retrieve your transaction history from the blockchain. For more details see bitcoin's BIP32 and BIP44.

**Comment by megmaltese** [comment]
Thank you very much for the answer!

I understand this, what I don't understand is how that initial key can
store future added addresses where I receive more funds.

"There is another formula that changes the previous private key into
another, seemingly random private key. This generates the second, third,
etc. private key."

Is this the part in question? That it's a chain of keys based on the
initial seed given by the worked 12 words?

One last thing, none of my business, but I would really like to know how
organizations like your can survive as you put no advertising in your
product.
Where do you make some money?

Thanks again for this great wallet and your work!


Marco E. G. Maltese

**Comment by dabura667** [comment]
Exactly.

--------

As far as revenue goes, I think bitpay makes most of its money from merchant services.

Copay and related libraries are profitable under the logic: "the easier for people to use bitcoin, the more customers buy from our merchants"

I don't work for them, but it's my guess.

**Comment by megmaltese** [comment]
Thank you very much for the answers!!



Marco E. G. Maltese
